### PR TITLE
Add typehints

### LINKED ISSUE #72: Missing Type Annotations
There are missing type annotations for some functions in this library.

The `typing` module does not exist on CircuitPython devices so the import needs to be wrapped in try/except to catch the error for missing import. There is an example of how that is done here:

```python
try:
    from typing import List, Tuple
except ImportError:
    pass
```

Once imported the typing annotations for the argument type(s), and return type(s) can be added to the function signature. Here is an example of a function that has had this done already:

```python
def wrap_text_to_pixels(
    string: str, max_width: int, font=None, indent0: str = "", indent1: str = ""
) -> List[str]:
```
If you are new to Git or Github we have a guide about contributing to our projects here: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

There is also a guide that covers our CI utilities and how to run them locally to ensure they will pass in Github Actions here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code In particular the pages: `Sharing docs on ReadTheDocs` and `Check your code with pre-commit` contain the tools to install and commands to run locally to run the checks.

If you are attempting to resolve this issue and need help, you can post a comment on this issue and tag both @foamyguy and @kattni or reach out to us on Discord: https://adafru.it/discord in the `#circuitpython-dev` channel.

The following locations are reported by mypy to be missing type annotations:

- [ ] adafruit_featherwing/led_segments.py:29
- [ ] adafruit_featherwing/led_segments.py:44
- [ ] adafruit_featherwing/led_segments.py:56
- [ ] adafruit_featherwing/led_segments.py:78
- [ ] adafruit_featherwing/led_segments.py:90
- [ ] adafruit_featherwing/auto_writeable.py:30
- [ ] adafruit_featherwing/pixelmatrix.py:33
- [ ] adafruit_featherwing/pixelmatrix.py:48
- [ ] adafruit_featherwing/pixelmatrix.py:57
- [ ] adafruit_featherwing/pixelmatrix.py:83
- [ ] adafruit_featherwing/pixelmatrix.py:100
- [ ] adafruit_featherwing/pixelmatrix.py:115
- [ ] adafruit_featherwing/pixelmatrix.py:130
- [ ] adafruit_featherwing/pixelmatrix.py:147
- [ ] adafruit_featherwing/pixelmatrix.py:170
- [ ] adafruit_featherwing/tft_featherwing.py:34
- [ ] adafruit_featherwing/tempmotion_featherwing.py:29
- [ ] adafruit_featherwing/tempmotion_featherwing.py:51
- [ ] adafruit_featherwing/tempmotion_featherwing.py:64
- [ ] adafruit_featherwing/tempmotion_featherwing.py:72
- [ ] adafruit_featherwing/tempmotion_featherwing.py:80
- [ ] adafruit_featherwing/tempmotion_featherwing.py:94
- [ ] adafruit_featherwing/tempmotion_featherwing.py:103
- [ ] adafruit_featherwing/sevensegment_featherwing.py:28
- [ ] adafruit_featherwing/rtc_featherwing.py:30
- [ ] adafruit_featherwing/rtc_featherwing.py:35
- [ ] adafruit_featherwing/rtc_featherwing.py:41
- [ ] adafruit_featherwing/rtc_featherwing.py:47
- [ ] adafruit_featherwing/rtc_featherwing.py:59
- [ ] adafruit_featherwing/rtc_featherwing.py:83
- [ ] adafruit_featherwing/rtc_featherwing.py:102
- [ ] adafruit_featherwing/rtc_featherwing.py:112
- [ ] adafruit_featherwing/rtc_featherwing.py:125
- [ ] adafruit_featherwing/rtc_featherwing.py:148
- [ ] adafruit_featherwing/rtc_featherwing.py:182
- [ ] adafruit_featherwing/rtc_featherwing.py:193
- [ ] adafruit_featherwing/rtc_featherwing.py:207
- [ ] adafruit_featherwing/rtc_featherwing.py:221
- [ ] adafruit_featherwing/rtc_featherwing.py:238
- [ ] adafruit_featherwing/rtc_featherwing.py:252
- [ ] adafruit_featherwing/rtc_featherwing.py:266
- [ ] adafruit_featherwing/rtc_featherwing.py:280
- [ ] adafruit_featherwing/rtc_featherwing.py:306
- [ ] adafruit_featherwing/neopixel_featherwing.py:28
- [ ] adafruit_featherwing/neopixel_featherwing.py:44
- [ ] adafruit_featherwing/neopixel_featherwing.py:77
- [ ] adafruit_featherwing/minitft_featherwing.py:53
- [ ] adafruit_featherwing/minitft_featherwing.py:84
- [ ] adafruit_featherwing/matrix_featherwing.py:30
- [ ] adafruit_featherwing/matrix_featherwing.py:40
- [ ] adafruit_featherwing/matrix_featherwing.py:47
- [ ] adafruit_featherwing/matrix_featherwing.py:62
- [ ] adafruit_featherwing/matrix_featherwing.py:82
- [ ] adafruit_featherwing/matrix_featherwing.py:95
- [ ] adafruit_featherwing/matrix_featherwing.py:104
- [ ] adafruit_featherwing/matrix_featherwing.py:113
- [ ] adafruit_featherwing/matrix_featherwing.py:122
- [ ] adafruit_featherwing/matrix_featherwing.py:141
- [ ] adafruit_featherwing/matrix_featherwing.py:153
- [ ] adafruit_featherwing/joy_featherwing.py:33
- [ ] adafruit_featherwing/joy_featherwing.py:160
- [ ] adafruit_featherwing/joy_featherwing.py:203
- [ ] adafruit_featherwing/ina219_featherwing.py:27
- [ ] adafruit_featherwing/gps_featherwing.py:28
- [ ] adafruit_featherwing/gps_featherwing.py:60
- [ ] adafruit_featherwing/gps_featherwing.py:72
- [ ] adafruit_featherwing/dotstar_featherwing.py:28
- [ ] adafruit_featherwing/alphanum_featherwing.py:28
- [ ] adafruit_featherwing/tft_featherwing_35.py:33
- [ ] adafruit_featherwing/tft_featherwing_24.py:33
- [ ] adafruit_featherwing/keyboard_featherwing.py:39

### PR DESCRIPTION
**Pull request by tekktrik**
Shout out to whoever added those Sphinx markups, you made life much easier!

Closes: #72

### CODE REVIEW

**Comment by tekktrik** [comment]
Meant to tag this as resolving issue #72

**Comment by jepler** [comment]
> Meant to tag this as resolving issue #72

I added text to the initial comment so that the issue would be linked.

**Comment by tekktrik** [comment]
Missed that, thanks!

**Comment by tekktrik** [comment]
I know there was a patch for regarding upgrading pylint, but I can't seem to get it, is that something I should do for this?

**Comment by kattni** [comment]
@tekktrik You don't need to do anything to upgrade Pylint on this repo, it happens behind the scenes. I've rerun the jobs - a number of the failures are a check we are disabling separately. I will rerun the jobs after that, and then you are welcome to take care of the rest of the linting issues in the repo (as per https://github.com/adafruit/Adafruit_CircuitPython_FeatherWing/pull/75/files) and we can close Dylan's PR in place of this one. Otherwise, I need to merge Dylan's PR, and then you have to update this one to match anyway. 

Are you up for linting this? You would need to update the imports, and disable `,duplicate-code` exactly like I explained on Discord for the other repo. Please ignore the `unspecified-encoding` check, as it should be disabled globally in a bit here.

**Comment by tekktrik** [comment]
Definitely up for linting!

**Comment by tekktrik** [comment]
Just merged that linting branch into mine

**Comment by tekktrik** [comment]
I see where to add the `,duplicate-code` comment, is it because the PATCH failed?

**Comment by tekktrik** [comment]
Nevermind that, working on it now!

**Comment by tekktrik** [comment]
Changes from #75 incorporated!

File: examples/featherwing_tft35_simpletest.py [hunk 1]
```diff
@@ -13,13 +13,15 @@
 tft_featherwing = tft_featherwing_35.TFTFeatherWing35()
 
 try:
-    f = open("/sd/tft_featherwing.txt", "w")
-    f.write("Blinka\nBlackberry Q10 Keyboard")
-    f.close()
+    with open(  # pylint: disable=unspecified-encoding
```
Comment (kattni) [inline examples/featherwing_tft35_simpletest.py]: Please remove this `pylint: disable=` as we've added it globally. Thanks!

File: examples/featherwing_tft35_simpletest.py [hunk 2]
```diff
@@ -13,13 +13,15 @@
...
+        "/sd/tft_featherwing.txt", "w"
+    ) as f:
+        f.write("Blinka\nBlackberry Q10 Keyboard")
 
-    f = open("/sd/tft_featherwing.txt", "r")
-    print(f.read())
-    f.close()
+    with open(  # pylint: disable=unspecified-encoding
```
Comment (kattni) [inline examples/featherwing_tft35_simpletest.py]: Same!

**Review by kattni — CHANGES_REQUESTED** [review]
Thanks so much! One more thing below.

**Comment by tekktrik** [comment]
Oh nice catch!  Removed!

**Review by kattni — APPROVED** [review]
This is great! Thanks for sticking through it, @tekktrik!

**Comment by tekktrik** [comment]
My pleasure!  Thanks for all the assistance on these!

**Pull request merged**